apps/system/test/unit/sync_manager_test.js

```diff
@@ -940,4 +963,248 @@ suite('system/SyncManager >', () => {
+        assert.equal(requestSpy.getCall(0).args[1], ERROR_SYNC_APP_KILLED);
+        done();
+      });
+    });
+
+    test('Is not Sync app', done => {
+      isSyncApp = false;
+      killapp();
+      setTimeout(() => {
+        assert.ok(requestSpy.notCalled);
+        done();
+      });
+    });
+  });
+
+  suite('appterminated', () => {
+    var syncManager;
+    var isSyncAppStub;
+    var isSyncApp;
+
+    function appterminated() {
+      window.dispatchEvent(new CustomEvent('appterminated', {
+        detail: {
+          origin: 'whatever'
+        }
+      }));
+    }
+
+    suiteSetup(() => {
+      syncManager = BaseModule.instantiate('SyncManager');
+      syncManager.start();
+    });
+
+    suiteTeardown(() => {
+      syncManager.stop();
+    });
+
+    setup(() => {
+      isSyncAppStub = this.sinon.stub(syncManager, 'isSyncApp', () => {
+        return isSyncApp ? Promise.resolve() : Promise.reject();
+      });
+    });
+
+    teardown(() => {
+      isSyncAppStub.restore();
+    });
+
+    test('No port', done => {
+      syncManager._port = undefined;
+      appterminated();
+      setTimeout(() => {
+        assert.ok(isSyncAppStub.notCalled);
+        done();
+      });
+    });
+
+    test('Is Sync app', done => {
+      isSyncApp = true;
+      syncManager._port = {};
+      appterminated();
+      setTimeout(() => {
+        assert.isNull(syncManager._port);
+        done();
+      });
+    });
+
+    test('Is not Sync app', done => {
+      isSyncApp = false;
+      syncManager._port = {};
+      appterminated();
+      setTimeout(() => {
+        assert.ok(syncManager._port);
+        done();
+      });
+    });
+  });
+
+  suite('isSyncApp', () => {
+    var syncManager;
+    var realMozApps;
+    var connections = [];
+
+    suiteSetup(() => {
+      syncManager = BaseModule.instantiate('SyncManager');
+      syncManager.start();
+
+      realMozApps = navigator.mozApps;
+      var app = {
+        getConnections: function() {
+          return Promise.resolve(connections);
+        }
+      };
+
+      var onsuccess = {
+        set onsuccess(callback) {
+          setTimeout(() => {
+            callback({ target: { result: app } });
+          });
+        }
+      };
+
+      navigator.mozApps = {
+        getSelf: function() {
+          return onsuccess;
+        }
+      };
+    });
+
+    suiteTeardown(() => {
+      syncManager.stop();
+      navigator.mozApps = realMozApps;
+    });
+
+    setup(() => {
+      connections = [];
+    });
+
+    [{
+      test: 'No origin',
+      origin: undefined
+    }, {
+      test: 'No connections',
+      origin: 'origin'
+    }, {
+      test: 'No matching connection',
+      origin: 'origin',
+      connections: [{
+        keyword: 'keyword',
+        origin: 'anotherorigin'
+      }]
+    }].forEach(config => {
+      connections = config.connections;
+      test(config.test, done => {
+        syncManager.isSyncApp(config.origin).then(() => {
```

Comment:
Nit: You could save some lines of code using chai-as-expected here, so I think it would be something like `expect(syncManager.isSyncApp(config.origin)).to.be.rejectedWith(undefined).and.notify(done);`